pkg/scale/scale.go

```diff
@@ -0,0 +1,93 @@
+package scale
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"sync"
+)
+
+// package level cache for fieldScaleIndicies
+var cache = &fieldScaleIndicesCache{
+	cache: make(map[string]fieldScaleIndices),
+}
+
+// fieldScaleIndex is used to map field index to scale index
+type fieldScaleIndex struct {
+	fieldIndex int
+	scaleIndex *string
+}
+type fieldScaleIndices []fieldScaleIndex
+
+// fieldScaleIndic
+type fieldScaleIndicesCache struct {
+	cache map[string]fieldScaleIndices
+	sync.RWMutex
+}
+
+func (fsic *fieldScaleIndicesCache) fieldScaleIndices(in interface{}) (v reflect.Value, indices fieldScaleIndices, err error) {
+	t := reflect.TypeOf(in)
+	v = reflect.ValueOf(in)
+	key := fmt.Sprintf("%s.%s", t.PkgPath(), t.Name())
+	if key != "." {
```

Comment:
is this case needed? is there a chance the key would be `"."`? if so add an else case

Reply:
yea I don't think so.  I'll return an `err` if I can't generate a key.  I'll test with anonymous structs to see if causes a problem.

Reply:
added tests for this.